core/new-gui/src/app/workspace/component/left-panel/time-travel/time-travel.component.ts

```diff
@@ -0,0 +1,120 @@
+import { Component, OnDestroy, OnInit } from "@angular/core";
+import { UntilDestroy, untilDestroyed } from "@ngneat/until-destroy";
+import { WorkflowActionService } from "../../../service/workflow-graph/model/workflow-action.service";
+import { WorkflowExecutionsEntry } from "../../../../dashboard/user/type/workflow-executions-entry";
+import { ExecuteWorkflowService } from "../../../service/execute-workflow/execute-workflow.service";
+import { WorkflowVersionService } from "../../../../dashboard/user/service/workflow-version/workflow-version.service";
+import {
+  WORKFLOW_EXECUTIONS_API_BASE_URL,
+  WorkflowExecutionsService,
+} from "../../../../dashboard/user/service/workflow-executions/workflow-executions.service";
+import { HttpClient } from "@angular/common/http";
+import { Observable } from "rxjs";
+import { map } from "rxjs/operators";
+
+const FULL_REPLAY_FLAG = "Full Replay";
+@UntilDestroy()
+@Component({
+  selector: "texera-time-travel",
+  templateUrl: "time-travel.component.html",
+  styleUrls: ["time-travel.component.scss"],
+})
+export class TimeTravelComponent implements OnInit, OnDestroy {
+  interactionHistories: { [eid: number]: string[] } = {};
+  public executionList: WorkflowExecutionsEntry[] = [];
+  expandedRows = new Set<number>(); // Tracks expanded rows by execution ID
+  public reverted = false;
+  public wid: number | undefined;
+  private timer: NodeJS.Timer | undefined;
+
+  constructor(
+    private workflowActionService: WorkflowActionService,
+    public executeWorkflowService: ExecuteWorkflowService,
+    private workflowVersionService: WorkflowVersionService,
+    private workflowExecutionsService: WorkflowExecutionsService,
+    private http: HttpClient
+  ) {}
+
+  ngOnInit(): void {
+    this.wid = this.workflowActionService.getWorkflowMetadata()?.wid;
+    if (this.wid === undefined) {
+      return;
+    }
+    // gets the versions result and updates the workflow versions table displayed on the form
+    let localWid = this.wid;
+    this.displayExecutionWithLogs(localWid);
+    this.timer = setInterval(() => {
+      this.displayExecutionWithLogs(localWid);
+    }, 5000); // trigger per 5 secs
```

Comment:
I want to make sure that this timer is properly destructured when the TimeTravel Component is destructed. In other components, we use `subscription` and `UntilDestroy()` hooks to ensure a subscription is unsubscribed. Can we use the same instead of "setInterval"?

Reply:
give me an example of using `subscription` to do a call periodically?

Reply:
found it.

Reply:
use https://rxjs.dev/api/index/function/interval ？